src/toolchain.rs

```diff
@@ -14,12 +14,21 @@ use util;
 
 const RUSTUP_BASE_URL: &'static str = "https://static.rust-lang.org/rustup/dist";
 
+const RUST_CI_TRY_BASE_URL: &'static str = "https://rust-lang-ci.s3.amazonaws.com/rustc-builds-try";
+
+const RUST_CI_COMPONENTS: [(&'static str, &'static str); 3] =
+    [
+        ("rustc", "rustc-nightly-x86_64-unknown-linux-gnu.tar.gz"),
+        ("rust-std-x86_64-unknown-linux-gnu", "rust-std-nightly-x86_64-unknown-linux-gnu.tar.gz"),
+        ("cargo", "cargo-nightly-x86_64-unknown-linux-gnu.tar.gz"),
```

Comment:
The single `rust-nightly` tarball would also work here as it contains all these components. Not sure if one approach is better.